File: 477. Total Hamming Distance/Java/Solution.java
public class Solution {
    public int totalHammingDistance(int[] nums) {
        int result = 0;
        for (int i = 0; i < 32; i++) {
            int base = 1 << i;
            int countOne = 0;
            for (int num : nums) {
                if ((base & num) != 0) {
                    countOne++;
                }
            }
            result += countOne * (nums.length - countOne);
        }
        return result;
    }
}

File: 477. Total Hamming Distance/C++/Solution.cpp
class Solution {
public:
    int totalHammingDistance(vector<int> &nums) {
        int result = 0;
        for (int i = 0; i < 32; i++) {
            int base = 1 << i;
            int countOnes = 0;
            for (int &num : nums) {
                if (base & num) {
                    countOnes++;
                }
            }
            result += countOnes * (nums.size() - countOnes);
        }
        return result;
    }
};